repository/src/main/java/ru/karalionak/rest/dao/CertificateDao.java

```diff
@@ -2,34 +2,22 @@
 
 
 import ru.karalionak.rest.entity.Certificate;
-import ru.karalionak.rest.query.SortParamContext;
+import ru.karalionak.rest.entity.CertificateDto;
+import ru.karalionak.rest.query.OrderedViewQuery;
 
 import java.util.List;
-import java.util.Map;
 import java.util.Optional;
 
-public interface CertificateDao {
+public interface CertificateDao extends Dao<Certificate> {
     void create(Certificate certificate);
 
-    void update(long id, Map<String, Object> updatedColumns);
+    void update(Certificate certificate);
 
-    void createCertificateTagRef(long certificateId, long tagId);
+    void createCertificateTagRef(long certificateId, List<Long> tagIds);
 
-    void delete(long id);
+    List<Certificate> findCertificateByTagId(long tagId);
 
-    Optional<Certificate> findById(long id);
-
-    List<Certificate> getAll();
-
-    List<Certificate> getAllWithSorting(SortParamContext sortParamContext);
-
-    List<Certificate> getAllWithFiltering(String pattern, List<Long> ids);
-
-    List<Certificate> getAllWithSortingAndFiltering(String pattern, List<Long> ids, SortParamContext sortParam);
-
-    List<Long> getTagIdsByCertificateId(long certificateId);
-
-    List<Long> getCertificateIdsByTagId(long tagId);
+    List<CertificateDto> findCertificateWithTags(OrderedViewQuery orderedViewQuery);
```

Comment:
Please, don't use such DTO's.
You can add tags exactly in Certificate and populate them always id dao or in services via dao.